tests/ProjectCodeTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Webmozart\Assert\Tests;
+
+use PHPUnit_Framework_TestCase;
+use ReflectionClass;
+use ReflectionMethod;
+
+/**
+ * @coversNothing
+ */
+class ProjectCodeTest extends PHPUnit_Framework_TestCase
+{
+    private static $readmeContent;
+    private static $assertDocComment;
+
+    public static function setUpBeforeClass()
+    {
+        self::$readmeContent = file_get_contents(__DIR__ . '/../README.md');
+
+        $rc = new ReflectionClass('\Webmozart\Assert\Assert');
+        self::$assertDocComment = $rc->getDocComment();
+    }
+
+    /**
+     * @dataProvider providesMethodNames
+     *
+     * @param string $method
+     */
+    public function testHasNullOr($method)
+    {
+        if ($method === 'notNull' || $method === 'null') {
```

Comment:
Technically we do offer `nullOr` for these two methods, but they make absolutely no sense, so i chose to not add them to the list

(No really when do you check for `nullOrNotNull` or `nullOrNull`?)